src/main/java/com/example/spring_boot_memo/csv/CsvService.java

```diff
@@ -0,0 +1,42 @@
+package com.example.spring_boot_memo.csv;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.csv.CsvGenerator;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import com.example.spring_boot_memo.csv.CsvData;
+@Service
+public class CsvService {
+
+    /**
+     * CsvMapperで、csvを作成する。
+     * @return csv(String)
+     * @throws JsonProcessingException
+     */
+
+
+    public String getCsvText(List<CsvData> csvDataList) throws JsonProcessingException {
+        CsvMapper mapper = new CsvMapper();
+        //文字列にダブルクオートをつける
+        mapper.configure(CsvGenerator.Feature.ALWAYS_QUOTE_STRINGS, true);
+        //ヘッダをつける
+        CsvSchema schema = mapper.schemaFor(CsvData.class).withHeader();
+        //本来はDBからデータを取得する。
+        return mapper.writer(schema).writeValueAsString(csvDataList);
```

Comment:
[Q]

```java
mapper.writer(schema).writeValueAsString(csvDataList)
```

ここの処理はなにをしていますでしょうか
何かWrite(書き込み処理が走っているように見えます)
getCsvTestメソッドはそのメソッド名でみると、データの更新はせず、データの取得のみが責務のメソッドに見えます。
書き込み処理があるのであれば、 WriteとReadでメソッドを別けてはどうでしょうか？

Reply:
分けました。
テストコード作成中でエラーが出たので報告します。とりあえず私が対処する予定です。
![image](https://user-images.githubusercontent.com/103587568/177092215-568bd97a-04e4-4e34-8d44-e4cc7ffa998b.png)

CsvServiceクラスのgetCsvHeaderメソッドだと推察します。
CsvSchema schema = mapper.schemaFor(CsvData.class).withHeader();
      
      java.lang.NullPointerException: Cannot invoke "com.fasterxml.jackson.dataformat.csv.CsvSchema.withHeader()" because the return value of "com.fasterxml.jackson.dataformat.csv.CsvMapper.schemaFor(java.lang.Class)" is null

Reply:
解決しました。
DIしてなかったのが原因でした。